pkg/api/query/v1.go

```diff
@@ -589,6 +591,16 @@ func (qapi *QueryAPI) query(r *http.Request) (interface{}, []error, *api.ApiErro
 	}
 
 	ctx := r.Context()
+
+	//var resultMetadataMutex sync.Mutex
+	//resultMetadata := storepb.SeriesStatsCounter{}
+	//resultMetadataReceiveFn := func(m storepb.SeriesStatsCounter) {
+	//	resultMetadataMutex.Lock()
+	//	defer resultMetadataMutex.Unlock()
+	//	resultMetadata = resultMetadata.CombineMetadata(m)
+	//}
+
+	//ctx = context.WithValue(ctx, seriesHeader, resultMetadataReceiveFn)
```

Comment:
This is the right approach here. You need a local variable for aggregated stats and update the variable in the callback function with a lock.